earthpy/clip.py

```diff
@@ -112,13 +112,13 @@ def clip_shp(shp, clip_obj):
     """
     try:
         shp.geometry
-    except AssertionError:
-        raise AssertionError("""Please provide a geoDataFrame with a valid
+    except AttributeError:
```

Comment:
These two try/except calls have a duplicate `AttributeError`. This could be DRY-ified as follows:

```python
try:
    shp.geometry
    clip_obj.geometry
except AttributeError:
    raise AttributeError("...")
```

Or, if you wanted to use different error messages to tell users explicitly that the `shp` or `clip_obj` input is invalid:

```python
try:
    shp.geometry
except AttributeError:
    # tell users the shp object needs a geometry column

try:
    clip_obj.geometry
except AttributeError:
    # tell users that clip_obj needs a geometry column
```

Also should `geoDataFrame` be `GeoDataFrame`?

Reply:
Totally. I"m going to combine them. I think the intent originally was to have custom messages but looking at it now...it's just as clear to combine.